src/components/v2/values/chartValuesDiff/ChartValuesView.tsx

```diff
@@ -920,9 +1006,49 @@ function ChartValuesView({
     const handleAppNameOnBlur = () => {
         if (commonState.activeTab === 'manifest') {
             updateGeneratedManifest(
+                isCreateValueView,
+                isExternalApp,
+                isDeployChartView,
+                appName,
+                valueName,
+                commonState,
+                commonState.chartValues.appStoreVersionId || commonState.chartValues.id,
+                commonState.modifiedValuesYaml,
+                dispatch,
+            )
+        }
+    }
+
+    const handleValueNameChange = (newValueName: string) => {
+        const validatedValueName = validationRules.appName(newValueName)
+        if (!validatedValueName.isValid && commonState.invalidAppNameMessage !== validatedValueName.message) {
+            dispatch({
+                type: ChartValuesViewActionTypes.multipleOptions,
+                payload: {
+                    invalidValueName: true,
+                    invalidValueNameMessage: validatedValueName.message,
+                },
+            })
+        } else if (validatedValueName.isValid) {
+            dispatch({
+                type: ChartValuesViewActionTypes.multipleOptions,
+                payload: {
+                    invalidValueName: false,
+                    invalidValueNameMessage: '',
+                },
+            })
+        }
+        setValueName(newValueName)
+    }
+
+    const handleValueNameOnBlur = () => {
```

Comment:
Can we re-use the `handleAppNameOnBlur` function by renaming it to `handleNameOnBlur` or something?

Reply:
Done